pandas/tseries/tests/test_base.py

```diff
@@ -45,6 +45,24 @@ def test_ops_properties_basic(self):
         self.assertEqual(s.day,10)
         self.assertRaises(AttributeError, lambda : s.weekday)
 
+    def test_shift(self):
```

Comment:
move this to ``tests/test_index`` (I make have told you to put in in ``tseries/test_base`` before, sorry), where the ``NotImplementedError`` is raised (this should go in the ``Base`` class ), then in ``Datetimelike`` you override to have it tested in the 3 datetimelikes